adrs/0001-pseudo-private-custom-properties.md

```diff
@@ -0,0 +1,28 @@
+# Refactor component styling using pseudo-private custom properties
+
+**Date:** [2022-11-03]
+
+### Issue
+Component styles are difficult to maintain. This is due to:
+
+1. Disjointed styles .less files amongst components. Within components, we currently sometime will output contextual, variant, modifier, interaction styles with the base component class and sometimes we don’t. These styles will might be in alphabetical order, by some logical order, in the order added, or seemingly randomly. Many component style .less files include a large preamble and table of contents that are seldom updated to reflect the current state of the component. 
```

Comment:
> Many component style .less files include a large preamble and table of contents that are seldom updated to reflect the current state of the component.

For my understanding: do PPCPs have self documenting characteristics as well?

Reply:
The way they're implemented in Stacks, yes. Specifically, the custom properties adhere to a naming convention to make them as clear as possible. They follow this format (all groups after the prefix separated by `-`):

- `--_` prefix to indicate it's a PPCP
- two-letter abbreviation of the component (if the component name is comprised of multiple words, I take the first letter of the first two words, otherwise user the first two letters of the component name)
- the child component name if applicable
- atomic class equivalent abbreviation or name if it exists, otherwise the logical name of the property
- interactive state if applicable

Examples:

- `--_ba-bc`: `border-color` on `.s-badge`
- `--_lp-footer-fd`: `flex-direction` on `.s-link-preview .s-link-preview--footer` 
- `--_ta-bg-hover`: `background-color` on `.s-tag` when hovering

I took this approach to make discerning the property's purpose obvious and to prevent collisions. I generally prefer self-documenting code with some support from short, inline comments over verbose code comments and find that in practice, large comments tend to get outdated often and can become a liability. Like most things, it depends, but that's my general feeling.

Should I include these details in the ADR or is that overkill?

Reply:
It just occurred to me that linking to [`lib/css/components/_styles-template.less`](https://github.com/StackExchange/Stacks/blob/develop/lib/css/components/_styles-template.less) would probably make sense here _**or**_ if we're ok with these ADRs being verbose, I could add the contents of that file here and remove the file itself. Thoughts?

Reply:
I vote to add it into the ADR in a language tagged code block. Imo, we should aim to keep unshippable code out of the core library folders as best we can (sometimes it's unavoidable, but it's my personal philosophy).

Something like the following:

~~~markdown
```less
.s-component-name {
    //...
}
```
~~~

Reply:
I agree with Ben. I wonder if there is an opportunity to remove the template file from that lib folder now that we have dedicated section for architecture decisions. 🤔